llvm/lib/Target/AMDGPU/AMDGPUCodeGenPrepare.cpp

```diff
@@ -1213,7 +1213,10 @@ Value *AMDGPUCodeGenPrepareImpl::expandDivRem24(IRBuilder<> &Builder,
                                                 BinaryOperator &I, Value *Num,
                                                 Value *Den, bool IsDiv,
                                                 bool IsSigned) const {
-  int DivBits = getDivNumBits(I, Num, Den, 9, IsSigned);
+  unsigned SSBits = Num->getType()->getScalarSizeInBits();
+  // If Num bits <= 24, assume 0 signbits.
+  unsigned AtLeast = (SSBits <= 24) ? 0 : (SSBits - 24);
```

Comment:
This changes from 9 sign bits to 8?

Reply:
I think I should take into account Den as well.

Reply:
For the 32-bit signed case, the 9 should be right? It's bit width - 1 for the sign bit? 8 for unsigned

Reply:
AtLeast is measuring number of sign bits and 9 would mean for i32, 23-bits are used in div/rem. So if computeNumSignBits returned 8, it would be disqualified. I think the code in expandDivRem24Impl should be able to handle 24-bit case though.

Reply:
Yes

Reply:
Oh, I missed the increment below. I think keeping it in the same expression would be easier to see, something like (...) + IsSigned